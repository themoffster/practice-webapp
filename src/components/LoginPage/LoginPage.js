import { Button, Form } from 'react-bootstrap'
import { LANDING } from 'src/enums/routes'
import { routeAndScrollTo } from 'src/helpers/router'
import React from 'react'

const LoginPage = () => {
  const [username, setUsername] = React.useState('')
  const [password, setPassword] = React.useState('')

  const onSubmit = async (event) => {
    event.preventDefault()
    routeAndScrollTo(LANDING)
  }

  return (
    <Form onSubmit={onSubmit}>
      <Form.Group className="mb-3" controlId="username">
        <Form.Label>Username</Form.Label>
        <Form.Control
          type="username"
          onChange={(e) => setUsername(e.target.value)}
        />
      </Form.Group>
      <Form.Group className="mb-3" controlId="password">
        <Form.Label>Password</Form.Label>
        <Form.Control
          type="password"
          onChange={(e) => setPassword(e.target.value)}
        />
      </Form.Group>
      <Button aria-label="Login" variant="primary" type="submit">
        Login
      </Button>
    </Form>
  )
}

export default LoginPage
