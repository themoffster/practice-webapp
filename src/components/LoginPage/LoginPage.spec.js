import { render } from '@testing-library/react'
import LoginPage from './LoginPage'
import React from 'react'

const getWrapper = () => render(<LoginPage />)

describe('LoginPage', () => {
  it('matches the snapshot', () => {
    const { getFragment } = getWrapper()
    expect(getFragment).toMatchSnapshot()
  })
})
