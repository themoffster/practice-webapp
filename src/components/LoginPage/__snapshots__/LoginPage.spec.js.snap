// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage matches the snapshot 1`] = `undefined`;
